packages/framework/collabspaces/src/test/collabSpaces.spec.ts

```diff
@@ -705,34 +940,56 @@ describe("Temporal Collab Spaces", () => {
 		};
 
 		const insertColsFn: Op = async (cp: IMatrix) => {
-			cp.insertCols(rand(cp.colCount), 1 + rand(3));
+			const pos = rand(Math.min(currentColCount, cp.colCount));
+			const count = 1 + rand(3);
+			cp.insertCols(pos, count);
+			currentColCount += count;
+			commandArray.push(
+				`insertColsFn post pos ${pos}, count ${count}, cp.colCount ${cp.colCount}  RowCount ${currentRowCount}, ColCount ${currentColCount}`,
+			);
 		};
 
 		const insertRowsFn: Op = async (cp: IMatrix) => {
-			cp.insertRows(rand(cp.rowCount), 1 + rand(3));
+			const pos = rand(Math.min(currentRowCount, cp.colCount));
```

Comment:
I do not think this is correct. In a sense that - this limits our test coverage.
It's totally normal for 2 containers to have different view on how many rows are there in container while they did not synchronize their changes. If this improves perf, then likely we have a bug in matrix that needs to be looked at. Based on what I understand, there should be no perf difference as Sequence (used as implementation for permutation vector) should not care and should have roughly same perf enumerating deleted segments as it does for "real" segments.
You might want to follow up with DanielL on that.

Reply:
I will revert it for now and track it separately as it does not affect passing/failures.